### ISSUE TITLE
Move source/ files in sub-folders according to their subject.

### ISSUE THREAD

**Issue by Bertram25** [OpenDungeons/OpenDungeons#10]
E.g: Move all the entities speaking files into a Entities/ sub-folder. Move the game map related files into a GameMap/ subfolder, …

A precise list of what should move is to be done first, though,

**Comment by Bertram25** [comment]
A proposal:
**In source/entities/:**
BattleField.cpp/h
Building.h
Creature.cpp/h
CreatureAction.cpp/h
CreatureDefinition.cpp/h
CreatureSound.cpp/h
DirectionalTrap.cpp/h
GameEntity.cpp/h
MapLight.cpp/h
MissileObject.cpp/h
ProximityTrap.cpp/h
Tile.cpp/h
Trap.h
TrapBoulder.h
TrapCannon.h
Weapon.cpp

**In source/goals:**
AllGoals.h
Goal.cpp/h
GoalClaimNTiles.cpp/h
GoalKillAllEnemies.cpp/h
GoalMineNGold.cpp/h
GoalProtectCreature.cpp/h
GoalProtectDungeonTemple.cpp/h

**In source/rooms:**
Room.cpp/h
RoomDojo.cpp/h
RoomDungeonTemple.cpp/h
RoomForge.cpp/h
RoomHatchery.cpp/h
RoomLibrary.cpp/h
RoomObject.cpp/h
RoomPortal.cpp/h
RoomQuarters.cpp/h
RoomTreasury.cpp/h

**in source/network:**
ChatMessage.cpp/h
ClientNotification.cpp/h
ODClient.cpp/h
ODPacket.cpp/h
ODServer.cpp/h
ODSocketClient.cpp/h
ODSocketServer.cpp/h
ServerNotification.cpp/h

**In source/modes:** (Everything directly used by application modes)
AbstractApplicationMode.cpp/h
Console.cpp/h
ConsoleMode.cpp/h
Console_getPromptCommand.cpp
Console_getHelp.cpp
EditorMode.cpp/h
FppMode.cpp/h
GameMode.cpp/h
InputManager.cpp/h
MenuMode.cpp/h
MenuModeMultiplayer.cpp/h
MenuModeSingleplayer.cpp/h
ModeManager.cpp/h
PreefixTree.cpp/h

**In source/gamemap:**
GameMap.cpp/h
MapLoader.cpp/h
MiniMap.cpp/h
TileContainer.cpp/h
TileCoordinateMap.cpp/h

**In source/ai:**
AIFactory.cpp/h
AIManager.cpp/h
AIWrapper.cpp/h
BaseAI.cpp/h
KeeperAI.cpp/h
NullAI.cpp/h

**In source/sound:**
SoundEffectsHelper.cpp/h
MusicPlayer.cpp/h

**In source/camera:** (The camera manager and its specific tool classes.)
CameraManager.cpp/h
CullingManager.cpp/h
CullingQuad.cpp/h
DummyArrayClass.cpp/h
FastPODAllocator.h
FixedPrecision.cpp/h
HermiteCatmullSpline.cpp/h
MortuaryQuad.cpp/h
RadialVector2.cpp/h
SlopeWalk.cpp/h

**In source/render:** (Everything more generic linked to rendering)
Gui.cpp/h
ODFrameListener.cpp/h
RenderManager.cpp/
RenderRequest.cpp/h
TextRenderer.cpp/h

**In source/utils:** (The most generic tools, or used in several locations.)
Helper.h
LogManager.cpp
Random.cpp/h
ResourceManager.cpp/h
Sleep.h
StackTracePrint.cpp/h
Vector3i.cpp/h

**Comment by akien-mga** [comment]
I'm 100% in favour of this. I'll let @hwoarangmy and @tomluchowski discuss the precise layout with you though, but your proposal sounds sensible.

**Comment by hwoarangmy** [comment]
Seems good. I just have 2 points :
- Tile is an entity
- I think the difference between render/camera/gamemap is tight as they heavily depend on each other. Moreover, RenderManager is more related to the gamemap/framelistener than to Gui

**Comment by Bertram25** [comment]
> Tile is an entity

True, will move it to entities/.

> I think the difference between render/camera/gamemap is tight as they heavily depend on each other.

True as well, but I had to make a choice and certain things are a matter of taste, right?

> Moreover, RenderManager is more related to the gamemap/framelistener than to Gui

Well, gui is heavily linked to stuff rendered IMO, and just fit there, as long as there is no more gui-related files. If it ever happens, I'd propose we move all the gui files in source/gui.

Ok with it?

**Comment by hwoarangmy** [comment]
Not sure there is something to change. As you said, things are mixed and you have to make choices. To be honest, I've not looked much in what is in the camera package. From the few I've seen, there was many public variables with short names and stuff like that that makes it really hard to follow so as long as I don't have to, I prefer avoiding them :)

**Comment by Bertram25** [comment]
> From the few I've seen, there was many public variables with short names and stuff like that that makes it really hard to follow so as long as I don't have to, I prefer avoiding them :)

Lol. Well, the camera code is IMHO a bit ill and need some love. I'm also overwhelmed by ALL the files and code needed to do some (unfortunately not working) culling here. Note that I don't say it's wrong, it's just fearsome O.O'
We might want to see whether there are simpler solutions to do that later.

**Comment by Bertram25** [comment]
Once, #25 will be done. I'll bite this one. ;)

**Comment by Bertram25** [comment]
Once my latest PR is done, I'll be on that one. (I've got a full evening tonight for that.)

**Comment by hwoarangmy** [comment]
> I've got a full evening tonight for that

No child to care of ? ;-)

**Comment by Bertram25** [comment]
> No child to care of ? ;-)

Not tonight, but tonight only. :)

**Comment by hwoarangmy** [comment]
Lucky you ^^
